src/core/meson.build

```diff
@@ -192,12 +192,14 @@ in_files = [['macros.systemd',   rpmmacrosdir],
 foreach item : in_files
         file = item[0]
         dir = item[1]
+        if install_sysconfdir or dir != pkgsysconfdir
+                configure_file(
+                        input : file + '.in',
+                        output : file,
+                        configuration : substs,
+                        install_dir : dir == 'no' ? '' : dir)
```

Comment:
I'd maybe prefer to fold the condition into `install_dir` here, something like:
```
install_dir = (install_sysconfdir or dir != pkgsysconfdir) and dir != 'no' ? dir : ''
cofnigure_file(
     ...
     install_dir : install_dir)
```
but current version is also OK.